components/movementsensor/wheeledodometry/wheeledodometry.go

```diff
@@ -436,50 +445,35 @@ func (o *odometry) DoCommand(ctx context.Context,
 
 	reset, ok := req[resetShift].(bool)
 	if ok {
-		if o.shiftPos {
-			o.position.X -= o.shiftX
-			o.position.Y -= o.shiftY
-		}
 		o.shiftPos = reset
-		o.shiftX = 0
-		o.shiftY = 0
+		o.originCoord = geo.NewPoint(0, 0)
+		o.position.X = 0
+		o.position.Y = 0
 		resp[resetShift] = fmt.Sprintf("resetting position and setting shift to %v", reset)
 	}
-	shift, ok := req[shiftPos].(bool)
-	if ok {
-		o.shiftPos = shift
-		if shift {
-			o.position.X += o.shiftX
-			o.position.Y += o.shiftY
-		} else {
-			o.position.X -= o.shiftX
-			o.position.Y -= o.shiftY
-		}
-
-		resp[shiftPos] = fmt.Sprintf("using setting position shift to %v", shift)
-	}
 	lat, okY := req[setLat].(float64)
 	long, okX := req[setLong].(float64)
 	if okY && okX {
-		geoOrigin := geo.NewPoint(0, 0)
-		shiftGeo := geo.NewPoint(lat, long)
-
-		distance := geoOrigin.GreatCircleDistance(shiftGeo) * 1000 // m
-		heading := rdkutils.DegToRad(geoOrigin.BearingTo(shiftGeo))
-		o.position.Y -= o.shiftY
-		o.shiftY = distance * math.Cos(heading)
-		o.position.Y += o.shiftY
-
-		o.position.X -= o.shiftX
-		o.shiftX = distance * math.Sin(heading)
-		o.position.X += o.shiftX
+		o.originCoord = geo.NewPoint(lat, long)
 		o.shiftPos = true
 
-		resp[setLat] = fmt.Sprintf("y position shifted to %v", o.position.Y)
-		resp[setLong] = fmt.Sprintf("x position shifted to %v", o.position.X)
-	} else {
+		resp[setLat] = fmt.Sprintf("lat shifted to %.8f", lat)
+		resp[setLong] = fmt.Sprintf("lng shifted to %.8f", long)
+	} else if okY || okX {
```

Comment:
should these both be `!ok`, hwen is this response meant to be triggered?

Reply:
hmmm, can a nil be sent over do command? Should we have nil checks there?

Reply:
this is meant to trigger when only one of the oks/values are given. 

- if both are set, record the new coord
- if only one is set, throw an error
- if neither are set, dont care

the previous behavior would always throw an error even if you were not shifting the positions

Reply:
Add a comment explaining that right under this line, then.